common/src/main/java/com/wynntils/core/chat/ChatModel.java

```diff
@@ -131,68 +130,59 @@ private static void handleMultilineMessage(Component message) {
         if (newLines.isEmpty()) {
             // No new lines has appeared since last registered chat line.
             // We could just have a dialog that disappeared, so we must signal this
-            handleNpcDialog(List.of());
+            handleNpcDialog(List.of(), true);
             return;
         }
 
-        if (newLines.getLast().isEmpty()) {
+        if (newLines.getLast().getString().isEmpty()) {
             // Wynntils add an empty line before the NPC dialog; remove it
             newLines.removeLast();
         }
 
-        LinkedList<String> newChatLines = new LinkedList<>();
+        LinkedList<Component> newChatLines = new LinkedList<>();
         LinkedList<String> dialog = new LinkedList<>();
 
-        if (NPC_FINAL_PATTERN.matcher(newLines.getFirst()).find()) {
+        String firstNewLineCoded = ComponentUtils.getCoded(newLines.getFirst());
+
+        boolean nonBlockingDialogue = false;
+
+        if (NPC_DIALOGUE_PATTERN.matcher(firstNewLineCoded).matches()) {
+            nonBlockingDialogue = true;
+            handleNpcDialog(List.of(firstNewLineCoded), false);
+        } else if (NPC_FINAL_PATTERN.matcher(firstNewLineCoded).find()) {
             // This is an NPC dialog screen.
             // First remove the "Press SHIFT to continue" trailer.
             newLines.removeFirst();
-            if (newLines.getFirst().isEmpty()) {
-                newLines.removeFirst();
-            } else {
-                WynntilsMod.warn("Malformed dialog [#1]: " + newLines.getFirst());
-            }
 
             // Separate the dialog part from any potential new "real" chat lines
-            boolean dialogDone = false;
-            for (String line : newLines) {
-                if (!dialogDone) {
-                    if (EMPTY_LINE_PATTERN.matcher(line).find()) {
-                        dialogDone = true;
-                        // Intentionally throw away this line
-                    } else {
-                        dialog.push(line);
-                    }
+            for (Component lineComponent : newLines) {
+                String line = ComponentUtils.getCoded(lineComponent);
+
+                if (NPC_DIALOGUE_PATTERN.matcher(line).find()) {
```

Comment:
Ooh, I did not spot this before. You cannot do this. Not all "NPC" dialogue lines are formatted like that. (That was why I kept the blank lines before, to be able to tell when the "NPC dialogue" lines were done. 

For an example, go and talk to the Detlas citizen standing next to the entrance from Nivla/Ragni. The dialogue ends like "You walk away, wondering if the citizen stands there all day greeting visitors" or something like that.

Reply:
Why? If we do not have a parsable quest, we set current dialogue to be empty. ...or is the problem that we don't parse the "you walk away.." part? It is not an actual dialogue though, is it?

Reply:
It is treated as part of the dialogue. You need to press shift to get past it. Go do Detlas and check with your code. I don't think it will work.